### PR TITLE
Make draggable callbacks check that artist has not been removed.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

Simpler solution to #10876 (see in particular https://github.com/matplotlib/matplotlib/pull/10876#issuecomment-376788351).

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/offsetbox.py [hunk 1]
```diff
@@ -1668,14 +1668,21 @@ def on_pick(self, evt):
...
+        if self._check_still_parented() and self.got_artist:
             self.finalize_offset()
             self.got_artist = False
             self.canvas.mpl_disconnect(self._c1)
 
             if self._use_blit:
                 self.ref_artist.set_animated(False)
 
+    def _check_still_parented(self):
+        if self.ref_artist.axes is None:
```
Comment (tacaswell) [inline lib/matplotlib/offsetbox.py]: should probably check `self.ref_artist.figure` too incase there are draggable things at the figure level.

**Comment by tacaswell** [comment]
:+1: I like this solution much better.

File: lib/matplotlib/offsetbox.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/offsetbox.py]: fixed

**Comment by anntzer** [comment]
Thoughts about moving away from using weakmethods in callbackregistry (see other thread for arguments)?  (note that that has a reasonable deprecation pathway: when we were going to remove a weakmethod from the registry, instead insert a proxy method that emits a warning the next time it's called)

**Comment by tacaswell** [comment]
Want to give more thought to changing away from weakrefs before taking a position, but my bias is to not change.

**Pull request merged**